src/storage.rs

```diff
@@ -61,26 +62,58 @@ where
                 .for_each(|name| trace!("{}", name.file_name().as_os_str().to_str().unwrap())); // @TODO handle unwrap explicitly
             self.init_from_existing(files_in_work_dir).unwrap(); // @TODO handle unwrap explicitly
         }
-        // @TODO implement
+        self.initialized = Some(());
         Ok(())
     }
 
-    /// Description
-    /// Examples
+    /// # Description
+    /// Writes bytes `value` to active blob
+    /// If active blob reaches it limit, create new and close old
+    /// Returns number of bytes, written to blob
+    /// # Examples
 
     // @TODO specify more useful error type
-    pub fn write(&mut self) -> Result<(), ()> {
-        // @TODO implement
-        Ok(())
+    pub fn write(&mut self, key: K, value: &[u8]) -> Result<usize, ()> {
+        self.initialized.ok_or(())?;
+        // @TODO process unwrap explicitly
+        if self.active_blob.as_ref().unwrap().size()? + value.len()
+            > self.config.max_blob_size.unwrap()
+            || self.active_blob.as_ref().unwrap().len()? >= self.config.max_data_in_blob.unwrap()
+        {
+            let new_active = Box::new(Default::default());
+            // @TODO process unwrap explicitly
+            let mut old_active = self.active_blob.replace(new_active).unwrap();
+            old_active.close()?;
+            self.blobs.push(*old_active);
+        }
+        // @TODO process unwrap explicitly
+        self.active_blob.as_mut().unwrap().write(&key, &value)
     }
 
-    /// Description
-    /// Examples
+    /// # Description
+    /// Reads data with given key to `Vec<u8>`, if error ocured or there are no
+    /// records with matching key, returns `Err(_)`
 
     // @TODO specify more useful error type
-    pub fn read(&self) -> Result<(), ()> {
-        // @TODO implement
-        Ok(())
+    pub fn read(&self, key: &K) -> Result<Vec<u8>, ()> {
+        // @TODO match error in map_err
+        if let Ok(data) = self
+            .active_blob
+            .as_ref()
+            .ok_or(())?
+            .read(key)
+            .map_err(|_e| debug!("no records in active blob"))
+        {
+            Ok(data)
+        } else if let Some(data) = self.blobs.iter().find_map(|blob| {
```

Comment:
Synchronization is required when accessing `blobs`. You can avoid it if you make `blobs` immutable and pointer sized.

Reply:
Rusts `Box<_>` is a pointer type, doc: [Box](https://doc.rust-lang.org/std/boxed/struct.Box.html)